src/edu/washington/escience/myria/operator/Operator.java

```diff
@@ -107,6 +108,30 @@ public ProfilingLogger getProfilingLogger() {
   }
 
   /**
+   *
+   * @return PythonFunctionRegistrar for operator.
+   * @throws DbException  in case of error.
+   */
+  public PythonFunctionRegistrar getPythonFunctionRegistrar() throws DbException {
+    PythonFunctionRegistrar pyFuncRegistrar = null;
+    try {
+      if (execEnvVars.get(MyriaConstants.EXEC_ENV_VAR_TEST_MODE) == null) {
+
+        if (getLocalSubQuery() instanceof WorkerSubQuery) {
+          pyFuncRegistrar =
+              ((WorkerSubQuery) getLocalSubQuery()).getWorker().getPythonFunctionRegistrar();
+        }
+      }
+
+    } catch (Exception e) {
+      LOGGER.debug("exception getting pyFuncRegistrar");
+      pyFuncRegistrar = null;
```

Comment:
I'm also confused, first this `e` should be a `DbException`, second, you don't need to try-catch if it's a `DbException`. Third, if you want to record a new error message, it's better to throw the new message (concatenated to the original one) than logging.